components/ParsePOSPayment/Receive-Invoice.tsx

```diff
@@ -37,16 +48,12 @@ function ReceiveInvoice({ recipientWalletCurrency, walletId, dispatch }: Props)
     return formatOperand(usdToSats(Number(amount)).toFixed().toString())
   }, [amount, currency, usdToSats])
 
-  const generateInvoice = React.useCallback(() => {
+  React.useEffect(() => {
```

Comment:
I'm a little confused by what this useEffect actually does?  It seems to create a new invoice every time the amount changes - is that right?

Reply:
Yes